array.go

```diff
@@ -913,3 +913,60 @@ func countElement(array []interface{}, element interface{}) int {
 	}
 	return count
 }
+
+// Filter accepts a function that returns a boolean. The function is ran
+// over the array items. If the function returns true, the item passes
+// the filter and is added to the new array of filtered items. If false,
+// the item is skipped (or in other words filtered out). After iterating
+// through all the items of the original array, the new filtered array
+// is returned.
+//
+// If there are any failed assertions in the filtering function, the
+// item is omitted without causing chain failure.
+//
+// Example:
+//
+// object := make(map[string]interface{})
+// filteredObject := object.Filter(func(_ string, value *Value) bool {
+//	value.String().NotEmpty()                //fails on 1 and 2
+//  return value != "bar"                    //fails on "bar"
+// })
+//
+// filteredArray.Raw() == []interface{}{"foo"} //true
+
+func (a *Array) Filter(filter func(index int, value *Value) bool) *Array {
```

Comment:
A few problems here:

* example code should be indented with tab (see other methods in httpexpect)
* object variable should be httpexpect.Object, not map, I guess
* I think you forget to initialize map values
* this example should be for Array, not Object
* extra empty line between doc comment and method
* instead of `filteredArray.Raw() == []interface{}{"foo"}`, can you please provide an assertion using httpexpect? like, `filteredArray.Equal([]interface{}{"foo"})` - like we do in other documentation examples

Reply:
Indentation inside example code is broken:

![image](https://user-images.githubusercontent.com/8436629/204821412-c96e9f8f-f44a-45ad-b563-fc2d8aa6bfd5.png)
****